src/provider/states/pod/starting.rs

```diff
@@ -79,41 +81,46 @@ async fn start_service_units(
             );
         }
 
-        add_annotation(
-            &client,
-            pod,
+        let mut annotations = HashMap::new();
+        annotations.insert(
             "featureLogs",
-            &systemd_service.invocation_id().await.is_ok().to_string(),
-        )
-        .await?;
+            systemd_service.invocation_id().await.is_ok().to_string(),
+        );
+        annotations.insert(
```

Comment:
The way I understand this code is that it tries to retrieve the invocation id and the restart count once after starting the container and if this attempt results in an error it will set  annotations accordingly.
Is there any chance of a race condition causing one or both of these calls to fail and causing a false negative in the annotations?
Happy for this to go in as is, just a thought that came to me when reading this.

Reply:
The two calls are theoretically independent (*) and there can be no race condition between them. There can be no false negatives as long as the connection to systemd is stable. Therefore I consider this to be a reliable solution.

I also considered to gather the systemd capabilities when the agent starts. As there is no service owned by the agent, I would have to find a well-known service to gather the properties or I would have to check the systemd version. The [documentation](https://www.freedesktop.org/software/systemd/man/org.freedesktop.systemd1) states that the version property should not be used for this case:
> `Version` encodes the version string of the running systemd instance. Note that the version string is purely informational. It should not be parsed and one may not assume the version to be formatted in any particular way. We take the liberty to change the versioning scheme at any time and it is not part of the public API.

(*) Practically I encountered a race condition in `zbus` where two mutexes blocked each other. This race condition did not produce a false positive but blocked the call and prevented the pod from entering the `Ready` state. This was solved in a newer [commit](https://gitlab.freedesktop.org/dbus/zbus/-/commit/6cdfe48cda5e0bf7b0dd8675be7a84439678afa9) in its `main` branch.